llama_index/llms/portkey.py

```diff
@@ -0,0 +1,387 @@
+"""
+    Portkey intergation with Llama_index for enchanced monitoring
+"""
+from typing import Any, Optional, Sequence, Dict, Union, List, TYPE_CHECKING, Type
+
+from llama_index.llms.custom import CustomLLM
+from llama_index.llms.base import (
+    ChatMessage,
+    LLMMetadata,
+    ChatResponse,
+    CompletionResponse,
+    ChatResponseGen,
+    llm_completion_callback,
+    llm_chat_callback,
+    CompletionResponseGen,
+)
+from llama_index.llms.portkey_utils import (
+    is_chat_model,
+    generate_llm_metadata,
+    get_llm,
+    IMPORT_ERROR_MESSAGE,
+)
+from llama_index.llms.generic_utils import (
+    completion_to_chat_decorator,
+    chat_to_completion_decorator,
+    stream_completion_to_chat_decorator,
+    stream_chat_to_completion_decorator,
+)
+
+from llama_index.bridge.pydantic import Field, PrivateAttr
+
+if TYPE_CHECKING:
+    from rubeus import (
+        Rubeus,
+        LLMBase,
+        RubeusModes,
+        RubeusCacheType,
+        RubeusCacheLiteral,
+        RubeusResponse,
+    )
+
+
+class Portkey(CustomLLM):
+    """_summary_
+
+    Args:
+        LLM (_type_): _description_
+    """
+
+    try:
```

Comment:
Sorry if it was confusing -- I think this should go into `__init__`

Reply:
Then, build/unit tests should pass

Reply:
Yes. That fixed it. Thanks :)